src/Maths/Silk.NET.Maths/Simd/Simd128.Load.cs

```diff
@@ -0,0 +1,318 @@
+                    }
+#endif
+                }
+                
+        
+                return Int64(ptr);
+            }            
+            [MethodImpl(Scalar.MaxOpt)]
+            static Vector128<T> Int64(T* ptr)
+            {
+                if (typeof(T) == typeof(long))
+                {
+#if AdvSIMD
+                    if (AdvSimd.IsSupported)
+                    {
+                        return (Vector128<T>) (object) AdvSimd.LoadVector128((long*) ptr);
+                    }
+#endif
+#if SSE
+                    if (Sse2.IsSupported)
+                    {
+                        return (Vector128<T>) (object) Sse2.LoadVector128((long*) ptr);
+                    }
+#endif
+#if AVX
+                    if (Avx2.IsSupported)
+                    {
+                        return (Vector128<T>) (object) Avx2.LoadVector128((long*) ptr);
+                    }
+#endif
+                }
+                
+        
+                return Single(ptr);
+            }            
+            [MethodImpl(Scalar.MaxOpt)]
+            static Vector128<T> Single(T* ptr)
+            {
+                if (typeof(T) == typeof(float))
+                {
+#if AdvSIMD
+                    if (AdvSimd.IsSupported)
+                    {
+                        return (Vector128<T>) (object) AdvSimd.LoadVector128((float*) ptr);
+                    }
+#endif
+#if SSE
+                    if (Sse2.IsSupported)
+                    {
+                        return (Vector128<T>) (object) Sse2.LoadVector128((float*) ptr);
+                    }
+#endif
+#if AVX
+                    if (Avx2.IsSupported)
+                    {
+                        return (Vector128<T>) (object) Avx2.LoadVector128((float*) ptr);
+                    }
+#endif
+                }
+                
+        
+                return Double(ptr);
+            }            
+            [MethodImpl(Scalar.MaxOpt)]
+            static Vector128<T> Double(T* ptr)
+            {
+                if (typeof(T) == typeof(double))
+                {
+#if AdvSIMD
+                    if (AdvSimd.IsSupported)
+                    {
+                        return (Vector128<T>) (object) AdvSimd.LoadVector128((double*) ptr);
+                    }
+#endif
+#if SSE
+                    if (Sse2.IsSupported)
+                    {
+                        return (Vector128<T>) (object) Sse2.LoadVector128((double*) ptr);
+                    }
+#endif
+#if AVX
+                    if (Avx2.IsSupported)
+                    {
+                        return (Vector128<T>) (object) Avx2.LoadVector128((double*) ptr);
+                    }
+#endif
+                }
+                
+        
+                return Other(ptr);
+            }
+            
+            [MethodImpl(Scalar.MaxOpt)]
+            static Vector128<T> Other(T* ptr)
+            {
+                            return Unsafe.ReadUnaligned<Vector128<T>>(ptr);
```

Comment:
What if T is not 32-bit?

Reply:
Why does it matter?

Reply:
Because you'd get a segfault. This method is meant to load a pointer containing 4 Ts, not a pointer containing a 128-bit vector.

Reply:
Oh, right. It's more than that, it should convert `ptr` to `Vector128<T>*`, that's why

Reply:
@Perksey we're reading a 128-bit vector, so the bitness of T doesn't matter

Reply:
@Perksey this method doesn't read 4 Ts it reads Vector128<T>.Count Ts. All methods operate on that count

Reply:
@HurricanKai ok fair enough.